app/Main.hs

```diff
@@ -42,10 +42,17 @@ getPlayer playerNumber gameState = case playerNumber of
                                         Player1 -> player1 gameState
                                         Player2 -> player2 gameState 
 
+flagPrompt :: IO Int
+flagPrompt = do
+    input <- ask "Please pick a flag"
+    let result = reads input :: [(Int, String)]
```

Comment:
So I wonder if maybe this can be thought of in a different way. For instance, you're converting a String -> Int -> Flag, but I wonder if you can skip the `Int` step and just convert a String directly into a Flag?